src/annotator/cross-frame.js

```diff
@@ -63,12 +63,26 @@ export class CrossFrame {
       frameIdentifiers.delete(frame);
     };
 
-    // Initiate connection to the sidebar.
-    discovery.startDiscovery((source, origin, token) =>
-      bridge.createChannel({ source, origin, token })
-    );
     frameObserver.observe(injectIntoFrame, iframeUnloaded);
 
+    /**
+     * Attempt to connect to the sidebar frame.
+     *
+     * @param {Window} frame
```

Comment:
Can you expand this to explain which frame we're talking about? I presume it's a given guest frame that wants to find a sidebar?